snuba/query/validation/signature.py

```diff
@@ -55,6 +64,16 @@ def __str__(self) -> str:
     Type[DateTime],
 ]
 
+AllowedScalarTypes = Union[
+    Type[None],
```

Comment:
Maybe you do not need the allow_nullable field anymore.

Reply:
I kept it because I personally prefer `Literal({int}, allow_nullable=True)` over `Literal({type(None), int})`. Something about taking the type of None gives me the willies. It also keeps it more consistent with the `Column` class.

Reply:
Then mind removing it from AllowedScalarTypes ? So it would be impossible to provide `Literal({None}, allow_nullable=False)` which allows the input to be nonsensical.